src/main/ruby/core/string.rb

```diff
@@ -643,21 +643,32 @@ def upto(stop, exclusive=false)
     stop = StringValue(stop)
 
     if stop.size == 1 && size == 1
+      enc = Encoding.compatible?(self.encoding, stop.encoding)
+
       return self if self > stop
       after_stop = stop.getbyte(0) + (exclusive ? 0 : 1)
       current = getbyte(0)
       until current == after_stop
-        yield current.chr
+        yield current.chr.force_encoding(enc)
```

Comment:
`Integer#chr` also takes an Encoding argument, which one should it be?

Reply:
I wasn't aware of that form. I'll make the switch.